localstack/services/awslambda/lambda_utils.py

```diff
@@ -307,7 +307,9 @@ class FilterCriteria(TypedDict):
     Filters: List[Dict[str, any]]
 
 
-def parse_and_apply_numeric_filter(record_value: Dict, numeric_filter: List[str | int]) -> bool:
+def parse_and_apply_numeric_filter(
+    record_value: Dict, numeric_filter: List[Union[str, int]]
```

Comment:
Sneaking in a small change that will help us with Python 3.9 compatibility issues in -platform. /cc @pandomic  Need to find a better solution for this, going forward.. :)